### DISCUSSION TITLE
Query to localhost uses the vite port instead of the baseUrl

### DISCUSSION THREAD

**Discussion by ghost** [reduxjs/redux-toolkit#3478] [Q&A]
I have a local backend running on localhost:8000, the api is fairly simple: 

```
import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react'
const localBackend = createApi({
    reducerPath: 'localBackend',
    baseQuery: fetchBaseQuery({
        baseUrl: 'http://localhost:8000/'
    }),

    endpoints: (builder) => ({
        scrapeImage: builder.query<string, string>({
            query: (url) => `img/${url}`,
        })
    }),
})


export default localBackend
```

however when I'm calling `const { data } = localBackend.useScrapeImageQuery(artist.url)` it fails with 404 because the request is made to: 

> GET http://localhost:5173/img/url 404 (Not Found)

With 5173 being the port of Vite where the app is running on. I tried all kind of localhost url variations without success, is there something I'm missing?

**Comment by markerikson** [comment]
That seems extremely odd, and shouldn't be happening based on the code you showed.  Can you show an example project where this happens?

**Reply by ghost** [reply]
I got it resolved and I doubt it's an issue with RTK. I can't pinpoint exactly what was causing it. My backend was very basic with Golang/mux, I switched to Gin because I already did Cors stuff there once. 
Also I'm sending an URL as parameter with the API request. I added an `encodeURIComponent` to it. 

So somewhere between the URL as parameter and the Go framework switch was an issue that caused this, my RTK api was unchanged. 

Maybe this can help point someone with a similar issue in the right direction. 

Edit: 

I tried to get a better diagnosis and created a sample repo. The missing **encodeURIComponent** was causing the issue. 
Without the request goes to the localhost where my react app is running, with the encodeURIComponent it targets my backend. 


Edit 2: 
While starting to investigate I found this SO post with the same issue, OP is also sending a raw URL with the query, apparently thats an issue.
https://stackoverflow.com/questions/73467636/rtk-query-is-htting-localhost-instead-of-baseurl

**Comment by DavOlufuwa** [comment]
Had this same issue recently. Still yet to get it resolved as I am using multiple Api slices in my RTK setup
